src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -31,35 +31,13 @@ private static void WorkerThreadStart()
                     // TODO: Event:  Worker thread wait event
                     while (s_semaphore.Wait(TimeoutMs))
                     {
-                        Volatile.Write(ref ThreadPoolInstance._separated.lastDequeueTime, Environment.TickCount);
                         if (TakeActiveRequest())
                         {
+                            Volatile.Write(ref ThreadPoolInstance._separated.lastDequeueTime, Environment.TickCount);
                             if (ThreadPoolWorkQueue.Dispatch())
                             {
-                                // If we ran out of work, we need to update s_separated.counts that we are done working for now
-                                // (this is already done for us if we are forced to stop working early in ShouldStopProcessingWorkNow)
-                                ThreadCounts currentCounts = ThreadCounts.VolatileReadCounts(ref ThreadPoolInstance._separated.counts);
-                                while (true)
-                                {
-                                    ThreadCounts newCounts = currentCounts;
-                                    newCounts.numProcessingWork--;
-                                    ThreadCounts oldCounts = ThreadCounts.CompareExchangeCounts(ref ThreadPoolInstance._separated.counts, newCounts, currentCounts);
-
-                                    if (oldCounts == currentCounts)
-                                    {
-                                        break;
-                                    }
-                                    currentCounts = oldCounts;
-                                }
-                                
-                                // It's possible that we decided we had no work just before some work came in, 
-                                // but reduced the worker count *after* the work came in.  In this case, we might
-                                // miss the notification of available work.  So we wake up a thread (maybe this one!)
-                                // if there is work to do.
-                                if (ThreadPoolInstance._numRequestedWorkers > 0)
-                                {
-                                    MaybeAddWorkingWorker();
-                                }
+                                // If the queue runs out of work for us, we need to update the number of working workers to reflect that we are done working for now
+                                MaybeRemoveWorkingWorker();
```

Comment:
I would just call this RemoveWorkingWorker, since it should always remove this worker by decrementing the count. Another worker may be added but any thread can satisfy that request.